### PR TITLE
Migrate download perm enforcement to data_permissions

### PR DESCRIPTION
**Pull request by noahmoss**
Updates download permission enforcement to read from the `data_permissions` table.

### CODE REVIEW

File: src/metabase/models/data_permissions.clj [hunk 1]
```diff
@@ -237,6 +237,34 @@
...
+                      :join [[:permissions_group :pg] [:= :pg.id :pgm.group_id]
+                             [:data_permissions :p]   [:= :p.group_id :pg.id]]
+                      :where [:and
+                              [:= :pgm.user_id user-id]
+                              [:= :p.perm_type (u/qualified-name :perms/download-results)]
+                              [:= :p.db_id database-id]]})
+          value-by-group
+          (-> (group-by :group_id perm-values)
+              (update-vals (fn [perms]
+                             (let [values (set (map :value perms))]
```
Comment (johnswanson) [inline src/metabase/models/data_permissions.clj:261]: I added a function above called `coalesce-less-restrictive`, which I think is doing the same thing you're doing here?

**Review by johnswanson — APPROVED** [review]

File: src/metabase/models/data_permissions.clj (hunk 1 above)
Comment (noahmoss) [reply src/metabase/models/data_permissions.clj:265]: Where did you add it? Don't see it on this branch

Comment (noahmoss) [reply src/metabase/models/data_permissions.clj:265]: oh it was `coalesce-most-restrictive`. updated thanks!

Comment (johnswanson) [reply src/metabase/models/data_permissions.clj:265]: Oops, sorry I missed your earlier message. 🤦

**Pull request merged**